vizro-core/src/vizro/models/types.py

```diff
@@ -299,10 +349,26 @@ def _parse_json(
             ) from exc
 
         import_path = json_schema_extra["import_path"]
+
+        # Check if we skip proper instantiation due to undefined function.
+        if function_name in (def_name for def_name, _ in allowed_undefined_captured_callables):
+            # Find the mode from allowed_undefined_captured_callables tuples
+            mode = next(mode for def_name, mode in allowed_undefined_captured_callables if def_name == function_name)
+            return CapturedCallable(function_name, mode=mode, **captured_callable_config)
```

Comment:
Doing this loop/conditional the other way round would make much more sense:

```suggestion
        # Check if we skip proper instantiation due to undefined function.
        for captured_callable in allowed_undefined_captured_callables:
            if function_name == captured_callable.function_name:
                return CapturedCallable(function_name, mode=captured_callable.mode, **captured_callable_config)
```

Reply:
N/A